### DISCUSSION TITLE
API to render glTF/GLB as image

### DISCUSSION THREAD

**Discussion by tehfonsi** [google/model-viewer#4911] [Show and tell]
We use `model-viewer` to provide an API to render images directly from a glTF or GLB file URL.

**How does it work?**
Imagine you have the URL of a glTF file, something like:

> https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/main/2.0/FlightHelmet/glTF/FlightHelmet.gltf

If you want to render this URL as an image, simply add the file URL to our API like this:

> https://www.flowkit.app/s/demo/r/https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/main/2.0/FlightHelmet/glTF/FlightHelmet.gltf

The image is rendered the first time you call the service, and afterward, it's served from a CDN.

There are a lot of use cases where this is useful, for example:

- Provide thumbnails of 3D files
- Compare images of 3D files in automated tests
- Ask questions about a 3D file using Open AIs vision model
- Render a 360-degree GIF of a 3D model
- Generate images for sharing on social media

It enables you to create automated workflows for things you had to do manually before.

You can try it and play around here: [https://www.flowkit.app/glb2png](https://www.flowkit.app/glb2png)
For more details, check the documentation: [https://docs.flowkit.app/](https://docs.flowkit.app/)

We’d love to hear your thoughts and feedback!

**Comment by elalish** [comment]
Very nice, thanks for sharing!
